lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,368 @@
+    ${['8', '9', '10', '11']}                           | ${'10'}       | ${'10'}
+    ${['8', '9', '10', '11']}                           | ${'4'}        | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'2020.04'}  | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'foobar'}   | ${null}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'bullseye'} | ${'bullseye'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'buster'}   | ${'buster'}
+    ${['jessie', 'stretch', 'buster', 'bullseye']}      | ${'warty'}    | ${null}
+    ${['jessie', 'stretch', 'buster', 'stable']}        | ${'stable'}   | ${'stable'}
+    ${['jessie', 'stretch', 'oldstable', 'bullseye']}   | ${'buster'}   | ${'oldstable'}
+    ${['jessie', 'oldoldstable', 'buster', 'bullseye']} | ${'warty'}    | ${null}
+  `(
+    'minSatisfyingVersion($versions, "$range") === "$expected"',
+    ({ versions, range, expected }) => {
+      expect(debian.minSatisfyingVersion(versions, range)).toBe(expected);
+    }
+  );
+
+  test.each`
+    currentValue      | rangeStrategy | currentVersion | newVersion    | expected
+    ${undefined}      | ${undefined}  | ${undefined}   | ${'foobar'}   | ${'foobar'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'11'}       | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'bullseye'} | ${'bullseye'}
+    ${'stretch'}      | ${undefined}  | ${undefined}   | ${'stable'}   | ${'bullseye'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'11'}       | ${'11'}
+    ${'oldoldstable'} | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'oldstable'}    | ${undefined}  | ${undefined}   | ${'11'}       | ${'stable'}
+    ${'9'}            | ${undefined}  | ${undefined}   | ${'stable'}   | ${'11'}
+  `(
+    'getNewValue("$currentValue", "$rangeStrategy", "$currentVersion", "$newVersion") === "$expected"',
+    ({ currentValue, rangeStrategy, currentVersion, newVersion, expected }) => {
+      expect(
+        debian.getNewValue({
+          currentValue,
+          rangeStrategy,
+          currentVersion,
+          newVersion,
+        })
+      ).toBe(expected);
+    }
+  );
+
+  test.each`
+    a                 | b                 | expected
+    ${'woody'}        | ${'sarge'}        | ${-1}
+    ${'lenny'}        | ${'3'}            | ${1}
+    ${'3'}            | ${'lenny'}        | ${-1}
+    ${'lenny'}        | ${'5'}            | ${0}
+    ${'squeeze'}      | ${'6'}            | ${0}
+    ${'10'}           | ${'buster'}       | ${0}
+    ${'6'}            | ${'squeeze'}      | ${0}
+    ${'buster'}       | ${'10'}           | ${0}
+    ${'oldoldstable'} | ${'8'}            | ${1}
+    ${'oldstable'}    | ${'oldoldstable'} | ${1}
+    ${'stable'}       | ${'oldstable'}    | ${1}
+    ${'11'}           | ${'oldoldstable'} | ${1}
+    ${'10'}           | ${'oldstable'}    | ${0}
+    ${'9'}            | ${'stable'}       | ${-1}
```

Comment:
make sure those tests won't simply fail when new Debian releases are added in future.

Reply:
now quite sure if possible, the rolling releases are translated to its numeric version and then compared. 
so as new releases are out, the numeric versions are changed as well.
so for example comparing `${'10'}           | ${'oldstable'}    | ${0}`  behind the scenes becomes a comparison between 10 and 11, as 12 becomes the new stable. the expected result in this case is `-1`

Reply:
We should mock the versions file or write some consts to easily update test from a single place.

Not sure what's the better solution

@rarkins WDYT?

Reply:
nice, const suggestion seems pretty straight forward and readable to me

```
    const STABLE_VER = '11'
    const OLDSTABLE_VER = '10'
    const OLDOLDSTABLE_VER = '9'
``` 
    ${STABLE_VER}                  | ${'oldoldstable'} | ${1}
    ${OLDSTABLE_VER}               | ${'oldstable'}    | ${0}
    ${OLDOLDSTABLE_VER}            | ${'stable'}       | ${-1}